sources/dynamodb/streaming_info.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package dynamodb
+
+import (
+	"sync"
+
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+)
+
+// StreamingInfo contains information related to processing of DynamoDB Streams.
+type StreamingInfo struct {
+	Records          map[string]map[string]int64 // Tablewise count of records received from DynamoDB Streams, broken down by record type i.e. INSERT, MODIFY & REMOVE.
+	recordsProcessed int64                       // Count of total records processed to Cloud Spanner(includes records which generated error as well).
+	ShardProcessed   map[string]bool             // Processing status of a shard, (default false i.e. unprocessed).
+	UserExit         bool                        // flag confirming if customer wants to exit or not, (false until user presses Ctrl+C).
+	Unexpecteds      map[string]int64            // Count of unexpected conditions, broken down by condition description.
+	lock             sync.Mutex
+}
+
+func MakeStreamingInfo() *StreamingInfo {
+	return &StreamingInfo{
+		Records:          make(map[string]map[string]int64),
+		recordsProcessed: int64(0),
+		ShardProcessed:   make(map[string]bool),
+		Unexpecteds:      make(map[string]int64),
+		UserExit:         false,
+		lock:             sync.Mutex{},
+	}
+}
+
+// makeRecordMaps initializes maps used to stores record count for
+// a given table.
+func (info *StreamingInfo) makeRecordMaps(srcTable string) {
+	info.Records[srcTable] = make(map[string]int64)
+}
+
+// SetShardStatus changes the processing status of a shard.
+//
```

Comment:
this line of comment got deleted unintentionally i guess

Reply:
actually this // is used to break lines here. It is a kind of printing "\n"